src/entrypoints/recorder.ts

```diff
@@ -420,13 +422,18 @@ function _tryReadBody(r: Request | Response): Promise<string> {
     // eslint-disable-next-line compat/compat
     return new Promise((resolve, reject) => {
         const timeout = setTimeout(() => resolve('[SessionReplay] Timeout while trying to read body'), 500)
-        r.clone()
-            .text()
-            .then(
-                (txt) => resolve(txt),
-                (reason) => reject(reason)
-            )
-            .finally(() => clearTimeout(timeout))
+        try {
+            r.clone()
+                .text()
+                .then(
+                    (txt) => resolve(txt),
+                    (reason) => reject(reason)
+                )
+                .finally(() => clearTimeout(timeout))
+        } catch (e) {
+            clearTimeout(timeout)
+            resolve('[SessionReplay] Failed to read body')
```

Comment:
so instead we should catch that error, and resolve with something meaningful, that means our captured network request would have this string as its body

and everything else should carry on

Reply:
it's not the only way to make this fail though.. it isn't _the_ bug